modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCart.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import AddToCartButton from './AddToCartButton';
+import WrapperWithQuantity from './WrapperWithQuantity';
+
+function AddToCart(props) {
+	return !props?.settings?.withQuantity ? (
+		<AddToCartButton {...props} />
+	) : (
+		<WrapperWithQuantity AddToCartButton={AddToCartButton} {...props} />
+	);
```

Comment:
This is a little confusing to read, not sure if it would make more sense to check for inverse
```suggestion
	return props?.settings?.withQuantity ? (
	        <WrapperWithQuantity AddToCartButton={AddToCartButton} {...props} />
	) : (
		<AddToCartButton {...props} />
	);
```

Reply:
Agreed!

TL;DR: In my mind I was more eager to check against the fewer and steadier _falsy_ possibilities to instantly decide we don't need a WrapperWithQuantity rather than the _truthy_ ones (`withQuantity` may exist, yet still not contain the appropriate props to start up render the component correctly). Psychological security I guess, nothing else, optional checks do the job just fine, as well as PropTypes in the QuantitySelector.